sources/packages/backend/apps/api/src/route-controllers/student/student.aest.controller.ts

```diff
@@ -464,4 +468,71 @@ export class StudentAESTController extends BaseController {
       );
     }
   }
+
+  /**
+   * Get possible matches for the student from the legacy system.
+   * @param studentId student ID to retrieve the data.
+   * @returns student legacy profile details.
+   */
+  @Roles(Role.AESTStudentLinkLegacyProfile)
```

Comment:
Why did we not consider a separate controller for the legacy endpoints? I feel that the legacy data is a different resource and even if otherwise the legacy operations can have a separate controller.